jms/1.7/modules/ROOT/pages/jms-activemq-configuration.adoc

```diff
@@ -1,153 +1,200 @@
+To configure these fields in Studio, follow these steps:
 
-`initialRedeliveryDelay`: In milliseconds, how much time to wait for the message to be redelivered the first time.
-`redeliveryDelay`: In milliseconds, how much time to wait until the message is subsequently redelivered after the first redelivery.
+. In Studio, open the *JMS Config* global element window.
+. In the *Connection Factory* section, set the *Factory configuration* field to *Edit inline*.
+. Set the *Initial redelivery delay* field to `1000`.
+. Set the *Redelivery delay* field to `100`.
+. Set the *Max redelivery* field to `10`.
+. Click *OK*.
+
+.JMS Connector Redelivery configuration
+image::jms-activemq-redelivery.png[Set the Max redelivery field to 10]
+
+In the XML editor, the `maxRedelivery`,`redeliveryDelay` and `initialRedeliveryDelay` configurations look like this:
 
 [source,xml,linenums]
 ----
 <jms:config name="JMS_Config">
   <jms:active-mq-connection>
-    <jms:factory-configuration
+    <jms:factory-configuration maxRedelivery="10"
       redeliveryDelay="100"
       initialRedeliveryDelay="1000"/>
   </jms:active-mq-connection>
 </jms:config>
 ----
 
-== Object Serialization
+== Configure Trusted Packages
+
+ActiveMQ versions 5.12.2, 5.13.0, and later restricts the classes that can be serialized and deserialized to prevent the execution of a malicious payload that can exploit the host system.
+
+For ActiveMQ, JMS Connector enables only the JDK and JRE provided classes by default, so if you
+need to exchange object messages, you must add the packages your applications use by activating the *Trusted packages* field. +
+Even though you can also enable the *Trust all packages* field to allow the serialization of more classes by allowing any object to be serialized and deserialized, this is not as secure as leaving it disabled in most cases. Keep this parameter disabled to improve the security and help prevent malicious attacks.
 
-ActiveMQ 5.12.2, 5.13.0, and later restricts the classes that can be serialized and deserialized to prevent the execution of a malicious payload that can exploit the host system.
+In the following example, you configure the connection to enable only users who are compliant with the `com.mulesoft.someapp` and `com.mulesoft.someapp.model` packages to consume and produce ObjectMessages:
 
-=== Trusted Packages
+. In Studio, open the *JMS Config* global element window.
+. In the *Connection Factory* section, set the *Factory configuration* field to *Edit inline*.
+. Select the *Trusted packages* field.
+. Click the plus sign to add a new package.
+. In the *Trusted package* window, set the *Value* field to `com.mulesoft.someapp`.
+. Click *Finish*.
+. Repeat Step 4.
+. In the *Trusted package* window, set the *Value* field to `com.mulesoft.someapp.model`.
+. Click *Finish*.
+. Select the *Trust all packages* field.
```

Comment:
Wait...what? Didn't we just tell them to not use this field?

Reply:
Maybe I don't understand something about the _combination_ of adding trusted packages and trusting all packages?